src/MoBi.Engine/Sbml/UnitDefinitionImporter.cs

```diff
@@ -31,16 +49,19 @@ public class UnitDefinitionImporter : SBMLImporter, IStartable, IUnitDefinitionI
       private readonly IMoBiDimensionFactory _moBiDimensionFactory;
       private readonly IDictionary<int, Unit> _baseUnitsDictionary;
       private readonly IDictionary<string, UnitConvertionInfo> _unitConvertionDictionary;
+      private IDimensionFactory _dimensionFactory;
+      public IReadOnlyDictionary<string, IDimension> ConvertionDictionary { get => _unitConvertionDictionary.ToDictionary(kv => kv.Key, kv => kv.Value.Dimension); }
       private IDictionary<string, string> _sbmlUnitsSynonyms = new Dictionary<string, string>()
```

Comment:
@abdelr Try to think Cache over Dictionary . cache exposes just so many more convenience methods
for instance you can remove the methods where you check if the value is in the cache completely by defining the behavior at the cache level

```
_sbmlUnitsSynonyms  = new Cache<string, string>(onMissingkeys = k=>k)
```
e.g what happens when the key is missing. we return the key.

Reply:
Check.